phonenumber_field/widgets.py

```diff
@@ -58,22 +68,30 @@ def __init__(self, attrs=None, initial=None):
         super().__init__(widgets, attrs)
 
     def decompress(self, value):
-        if value:
-            if type(value) == PhoneNumber:
-                if value.country_code and value.national_number:
-                    return [
-                        "+%d" % value.country_code,
-                        national_significant_number(value),
-                    ]
-            else:
-                return value.split(".")
-        return [None, ""]
+        region = getattr(settings, "PHONENUMBER_DEFAULT_REGION", None)
+
+        if isinstance(value, str):
+            value = to_python(value=value, region=region)
+
+        if isinstance(value, PhoneNumber):
+            if not value.is_valid():
+                region_code = getattr(value, "_region", region)
+                return [region_code, value.raw_input]
+            region_code = region_code_for_number(value)
+            national_number = str(value.national_number)
```

Comment:
I don’t think that is correct. Some national phone numbers start with a `0`, which is significant. `national_number` is an integer, so to know how many significant `0` should be included, one must look at the `number_of_leading_zeros` and `italian_leading_zero` properties.
That’s what the `national_significant_number` does.
https://github.com/daviddrysdale/python-phonenumbers/blob/75ca34bc7c889bc3c501d00b2b1f8ec4435cb821/python/phonenumbers/phonenumberutil.py#L1690

Reply:
I think you are right. I must have missed it.

Reply:
This is covered with 2a9a1d9.